python/ray/autoscaler/azure/node_provider.py

```diff
@@ -185,26 +193,29 @@ def create_node(self, node_config, tags, count):
                 e.args += ("name", vm_name)
                 raise
 
-            # get public ip address
-            public_ip_addess_params = {
-                "location": location,
-                "public_ip_allocation_method": "Dynamic"
+            ip_configuration = {
```

Comment:
is the goal to accelerate setup? I don't think it hurts to be able to hit the workers externally

Reply:
more about security. suddenly the surface is rather big. setup time is another factor

Reply:
there's a flag that can be set in the config (use_internal_ips), what if we check that and alter the behavior if necessary. i'm also happy to have it set to true in the default config